aws-rotate-iam-keys.template.rb

```diff
@@ -21,11 +21,15 @@ class AwsRotateIamKeys < Formula
   devel do
     Dir.chdir(File.expand_path(File.join(File.dirname(__FILE__), '../'))) do
       url %x{git config --local --get remote.origin.url | tr -d '\n'}, using: :git, branch: "develop"
-      version %x{git describe develop --always | tr -d '\n'}
+      version "head"
     end
   end
 
   def install
+    version = File.read('VERSION')
+    code = File.read("src/bin/aws-rotate-iam-keys")
+    code = code.gsub(/<%VERSION%>/, version)
+    File.open("src/bin/aws-rotate-iam-keys", "w") {|file| file.puts code}
```

Comment:
I haven't tested this, but I don't think it'll work as the current working directory when this code is run won't necessarily be the directory containing the `VERSION` file.

Reply:
Gotcha. I'll re-work this section and test it my different systems.

Reply:
I should have also asked what the purpose of this is. It looks like more magic/voodoo to rewrite source code on the fly, in this case to inject the correct version number into the script when it is installed via homebrew, but this is all so hard to follow. It works, but at the cost of making things very hard to understand - where is the true source?

Reply:
The source for this is the zipped files. There are numerous release artifacts in different formats that come out of this repo each time a new version is cut, including the state of the repo itself for the Homebrew installation. I'm considering scrapping the template stuff as well as this code change in favor of a folder change that puts all the source files in `src` and then the resulting files will be built to `Formula` for homebrew, `releases` for the deb and zip files, and `website` for the website. I think that will clear up any confusion on the code files. Then I'll come up with a `sed` replace for the README to get rid of that template. Thoughts?

Reply:
I don't think any build process should generate source files, it should just package them. 

Does the Homebrew archive really need to include the entire repo? I would have thought it just needs to contain the files needed to install the binary. 

The Homebrew formula is a source of confusion, because having a Homebrew formula included in the same repo as the binary it installs is unusual. However, I don't think you should try and change that just yet, small steps and all. If the source of the formula moves out of `Formula`, but there is still a formula file in `Formula`, that's going to be confusing.

I think it would less confusing to move the `aws-rotate-iam-keys` into a `bin` directory from the root, create a new `debian` directory at the root for the debian stuff, and treat all these files as actual source rather than build templates. Then you'd have a project root with `Formula`, 'Windows`, `bin`, `debian` and `website`, which seems pretty clear to me.

I might be simplifying, but building a release without templates seems like it should be easy enough... bump the version, generate the release archives, update the various checksums.

I don't know anything about debian packages, so maybe it makes complete sense for the files in the `debian` directory to be templates with an interpolated `VERSION` variable for building the debian/ubuntu packages. If templates aren't necessary, it seems like it would be simple to replace the use of build templates with `sed` (the `VERSION` variable is only used to set the `Version` and `Standard-Version` fields in the debian and ubuntu control files - p.s. should this really be used to set `Standard-Version`?).

Bumping the version in Homebrew formula should be simple using `sed`, and could be simple in the `aws-rotate-iam-keys` script itself by setting a `VERSION` variable at the top of the script rather than interpolating the `VERSION` into the help output. Updating the readme is admittedly a little trickier, but surely not that hard (and could be really simple if it was updated slightly to just avoid referencing specific version numbers).

Generating the release archives seems to work fine already, though the Homebrew archive doesn't need the entire repo, just the bin directory.

Updating the checksums is currently tricky because this also updates the readme, but it doesn't have to, those checksums could be separate files in the `releases` directory, one for each archive/package, and bumping the sha in the homebrew formula is easy with `sed`.

Sorry, gotta go, sorry if some of the above is long winded, but haven't time to summarise in a TLDR